Makefile

```diff
@@ -120,11 +120,11 @@ protos: $(PROTO_GOS)
 	protoc -I $(GOPATH)/src:./vendor:./$(@D) --gogoslick_out=plugins=grpc,Mgoogle/protobuf/any.proto=github.com/gogo/protobuf/types,:./$(@D) ./$(patsubst %.pb.go,%.proto,$@)
 
 lint:
-	./tools/lint -notestpackage -novet -ignorespelling queriers -ignorespelling Queriers .
-
 	# -stdmethods=false disables checks for non-standard signatures for methods with familiar names.
 	# This is needed because the Prometheus storage interface requires a non-standard Seek() method.
-	go vet -stdmethods=false ./pkg/...
+	go vet -tags netgo -stdmethods=false ./pkg/...
+	misspell -error docs
+	golangci-lint run --new-from-rev ed7c302fd968 --build-tags netgo --timeout=5m --enable golint --enable misspell --enable gofmt
```

Comment:
This is particularly slow when running on OSX (with Docker for Mac) due to the shared volumes performances. Do you see any problem adding `:delegated` to the 3 related volumes, switching

```
	@$(SUDO) time docker run $(RM) $(TTY) -i \
		-v $(shell pwd)/.cache:/go/cache \
		-v $(shell pwd)/.pkg:/go/pkg \
		-v $(shell pwd):/go/src/github.com/cortexproject/cortex \
		$(BUILD_IMAGE) $@;
```

to

```
	@$(SUDO) time docker run $(RM) $(TTY) -i \
		-v $(shell pwd)/.cache:/go/cache:delegated \
		-v $(shell pwd)/.pkg:/go/pkg:delegated \
		-v $(shell pwd):/go/src/github.com/cortexproject/cortex:delegated \
		$(BUILD_IMAGE) $@;
```

In my case, the execution time difference is:

- Current: `8m8.369s`
- With delegated: `1m38.258s`

Reply:
What does `:delegated` do?
Would we want it on all similar lines?

Reply:
> What does :delegated do?

The `:delegated` tells Docker for Mac that the container’s view is authoritative (permit delays before updates on the container appear in the host). Details here: https://docs.docker.com/docker-for-mac/osxfs-caching/

I don't have a Linux box right now to play with, but it should just be ignored on Linux.

> Would we want it on all similar lines?

I would say yes.

Reply:
Have you considered adding `:delegated` to volume mounts?

Reply:
Have done this now.